databricks/koalas/internal.py

```diff
@@ -359,6 +363,16 @@ def __init__(self, sdf: spark.DataFrame,
         :param column_index_names: Names for each of the index levels.
         """
         assert isinstance(sdf, spark.DataFrame)
+        if index_map is None:
+            # Here is when Koalas DataFrame is created directly from Spark DataFrame.
+            assert column_index is None
+            assert column_index_names is None
+
+            if "__index_level_0__" not in sdf.schema.names:
```

Comment:
I will make a separate PR to completely disallow no-index Koalas DataFrame. Seems like there are multiple places to fix.

One side question is that how we will handle the roundtrip in Koalas and Spark DataFrame. If Koalas has an index, `to_spark()` loses index information and we have nowhere to store.

I think, until we figure out a way to store index information properly, we should strip index when we convert to Spark DataFrame.

Currently it's kind of funny:

```
>>> import databricks.koalas as ks
>>> ks.DataFrame(ks.DataFrame({'a': [1,2,3]}).to_spark())
   __index_level_0__  a
0                  0  1
1                  1  2
2                  2  3
```

So basically I would like to propose below until we figure out a clever way to roundtrip.

```
>>> ks.DataFrame({'a': [1,2,3]}).to_spark().show()
+---+
|  a|
+---+
|  1|
|  2|
|  3|
+---+
```

Reply:
cc @ueshin and @rxin.

Reply:
But to do that, we should keep index mapping information somewhere. After converting Koalas DataFrame into Spark DataFrame, it's being lost.

Reply:
yeah, loosing the mapping should be okay, I just meant that if the index is named, then the name should be the column name for the Spark DataFrame.

Reply:
Oh, oh right. Yes.